WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Package Details/ShippingLabelPackageDetailsViewModel.swift

```diff
@@ -90,6 +94,9 @@ final class ShippingLabelPackageDetailsViewModel: ObservableObject {
         self.packagesResponse = packagesResponse
         self.selectedPackageID = selectedPackageID
         self.totalWeight = totalWeight ?? ""
+        if totalWeight != nil {
+            self.isPackageWeightEdited = true
+        }
```

Comment:
I'm not sure this is the expected behavior. I tested with the following steps:
- I select one of the boxes (custom or not) and tapped Done button
- After navigating back to the Create Shipping Label form - I get back and change to another box.
- Now my previous choice is deemed custom input (but it isn't), therefore the weight of my box isn't updated automatically when I choose any other box.

I tested the same steps on the web app and saw that as long as I don't manually input a weight, I can still get automatic weight updates when I get back.

I would suggest that you check for the initial `isPackageWeightEdited` by comparing the initial `totalWeight` with a calculated weight from the products, variations and initial package. If the weight is the same as calculated - we can safely decide that the initial weight was not input manually.

Reply:
You're right, thanks! I must have input the weight manually while testing the web behavior, and didn't think about the different expectation when the weight was saved but hadn't been input manually. I have a fix for that using your suggested approach in 2689621.